core/src/main/java/io/questdb/cairo/ColumnTypeDriver.java

```diff
@@ -176,6 +176,18 @@ void o3sort(
      */
     long setAppendAuxMemAppendPosition(MemoryMA auxMem, long rowCount);
 
+    /**
+     * Sets the append position in both the auxiliary and data vectors.
+     * <p>
+     * <strong>Warning:</strong> The expected usage of the returned value is to <strong>estimate</strong> the average size of a record.
+     * When a table is empty (i.e., the position is 0), there is no data available to estimate the size of a record, so this method returns
+     * a fixed constant instead of 0. Therefore, it should not be relied upon for precise calculations, such as truncating buffers, etc.
```

Comment:
What should be relied upon for precise calculation instead? Currently, this method is used in `TableWriter.setColumnAppendPosition`

Reply:
it's used like this:
```java
    private void setAppendPosition(final long rowCount, boolean doubleAllocate) {
        long recordLength = 0;
        for (int i = 0; i < columnCount; i++) {
            recordLength += setColumnAppendPosition(i, rowCount, doubleAllocate);
        }
        avgRecordSize = rowCount > 0 ? recordLength / rowCount : Math.max(avgRecordSize, recordLength);
    }
```
that's the only place where the return value is used - to calculate the average record size.

Reply:
If it sets the append position to 0 why does it return an approximate size and not 0?

Reply:
I can only guess the intent here was to estimate the average record size for a given table (with a given schema, etc). even the table has no row yet. 

OTOH: It's inconsistent with fixed-width column and there is an explicit `TableUtils.estimateAvgRecordSize(metadata)` anyway.

Reply:
Probably we should split this 2-in-1 method and add a utility method to `TableUtils` instead of the current weird behavior.

Reply:
plot twist: there already is such method! :)

Reply:
wdyt? it's still a bit convoluted due to the +1 entries with String columns.

Reply:
Still a bit weird, but better than it was. 😂